app/assets/javascripts/preferences/panes/account/subscription/subscription_state.tsx

```diff
@@ -0,0 +1,53 @@
+import { action, computed, makeObservable, observable } from 'mobx';
+
+type Subscription = {
+  planName: string;
+  cancelled: boolean;
+  endsAt: number;
+};
+
+type AvailableSubscriptions = {
+  [key: string]: {
+    name: string;
+  };
+};
+
+export class SubscriptionState {
+  userSubscription: Subscription | undefined = undefined;
+  availableSubscriptions: AvailableSubscriptions | undefined = undefined;
+
+  constructor() {
+    makeObservable(this, {
+      userSubscription: observable,
+      availableSubscriptions: observable,
+
+      userSubscriptionName: computed,
+
+      setUserSubscription: action,
+      setAvailableSubscriptions: action,
+    });
+  }
+
+  get userSubscriptionName(): string {
+    if (
+      this.availableSubscriptions &&
+      this.userSubscription &&
+      this.availableSubscriptions[this.userSubscription.planName]
+    ) {
+      return this.availableSubscriptions[this.userSubscription.planName].name;
+    }
+    return '';
+  }
+
+  public setUserSubscription(subscription: Subscription): void {
+    console.log('set subscription in state', subscription);
```

Comment:
Please remove this console statement and the one on line 45 :)

Reply:
Oops! Nice catch, thanks 🙂